tests/test_hnsw_new.cpp

```diff
@@ -0,0 +1,1177 @@
+
+// Copyright 2024-present the vsag project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <spdlog/spdlog.h>
+
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/generators/catch_generators.hpp>
+#include <fstream>
+#include <iostream>
+#include <limits>
+#include <nlohmann/json.hpp>
+#include <numeric>
+#include <random>
+
+#include "simd/simd.h"
+#include "test_index.h"
+#include "vsag/errors.h"
+#include "vsag/vsag.h"
+
+TEST_CASE_METHOD(fixtures::TestIndex, "HNSW Build & ContinueAdd Test", "[ft][hnsw]") {
```

Comment:
`test_hsnw_new.cpp` is used for new VSAG index framework test suite ?

Reply:
the test_hnsw_new will replace test_hnsw soon, it's from the basic index test, but only test features what hnsw need